src/shogun/mathematics/RandomMixin.h

```diff
@@ -0,0 +1,64 @@
+#ifndef __RANDOMMIXIN_H__
+#define __RANDOMMIXIN_H__
+
+#include <shogun/base/SGObject.h>
+#include <shogun/lib/config.h>
+
+#include <random>
+
+namespace shogun
+{
+	static inline void seed_callback(CSGObject*, int32_t);
+
+	template <typename Parent, typename PRNG = std::mt19937_64>
+	class RandomMixin : public Parent
+	{
+	public:
+		template <typename... T>
+		RandomMixin(T... args) : Parent(args...)
+		{
+			init();
+			m_seed = 0;
```

Comment:
this is wrong... this should use as in case of the example random_device... so that the PRNG is actually really RNG...

Reply:
An earlier implementation had a random_device but I removed it. I will bring it back.

Reply:
and actually the problem is that we should support array of seeds as well :DDD
because the PRNG to be really a RNG should be init-ed like this (with an array):
```
std::random_device source;
std::generate(std::begin(random_data), std::end(random_data), std::ref(source));
std::seed_seq seeds(std::begin(random_data), std::end(random_data));
PRNG(seeds);
```

Reply:
but for the time being that we figure out stuff, let's just use a single int32_t... but that should come from random_device

Reply:
ok good, basically if seed is not set it should be random